src/main/java/gov/cabinetofice/gapuserservice/service/OneLoginService.java

```diff
@@ -132,10 +215,13 @@ public Map<String, String> generateCustomJwtClaims(final OneLoginUserInfoDto use
         return claims;
     }
 
-    public OneLoginUserInfoDto getOneLoginUserInfoDto(final String code) {
+    public JSONObject getOneLoginUserTokenResponse(final String code) {
         final String oneLoginJwt = createOneLoginJwt();
-        final String authToken = getAuthToken(oneLoginJwt, code);
-        return getUserInfo(authToken);
+        return getTokenResponse(oneLoginJwt, code);
```

Comment:
Possibly a SSRF from this, the code looks to be passed directly from the controller to this function which does string concatenation - need to add validation and sanitization

Reply:
now sanitising using Jsoup:
```
    private String sanitizeCode(String code) {
        return Jsoup.clean(
                StringEscapeUtils.escapeHtml4(
                        StringEscapeUtils.escapeEcmaScript(StringUtils.replace(code, "'", "''"))
                ),
                Safelist.basic()
        );
    }
```